### ISSUE TITLE
[BUG] Click then type on an input, selection messed up

### ISSUE THREAD

**Issue by jperl** [microsoft/playwright#1331]
**Context:**
- Playwright Version: 0.11.1-next.1583883243818
- Operating System: Mac

**Code Snippet**

```javascript
const playwright = require("playwright");

(async () => {
  const browser = await playwright.chromium.launch({ headless: false });
  const page = await browser.newPage();
  await page.goto("https://m4tkx.csb.app");
  await page.click("#date-input");
  await page.type("#date-input", "02022020");
})();

```

**Describe the bug**

The click messes up the selection, and type does not reset it.

![2020-03-10 18 18 04](https://user-images.githubusercontent.com/1136652/76370209-bb7bfd80-62fb-11ea-821a-eb5cb2167463.gif)

**Comment by jperl** [comment]
Example test scenario: https://docusaurus.io search box lazily loads algolia after you click on the input.

**Comment by dgozman** [comment]
I looked at this scenario. The issue is that clicking on a date input selects year component, so typing there does not fill the right value. I think that is working as intended. This should not happen with a typical text or search input though, please file an issue if that does.

Focusing and typing works:
```js
await page.focus("#date-input");
await page.type("#date-input", "02022020");
```

Just typing works:
```js
await page.type("#date-input", "02022020");
```

I will also make fill work with dates:
```js
await page.fill("#date-input", "02022020");
```

Even after click:
```js
await page.click("#date-input");
await page.fill("#date-input", "02022020");
```

**Comment by jperl** [comment]
If fill worked for date inputs that would be great. 

Right now with playwright@0.12.1

```
const playwright = require("playwright");

(async () => {
  const browser = await playwright.chromium.launch({ headless: false });
  const page = await browser.newPage();
  await page.goto("https://m4tkx.csb.app");
  await page.fill("#date-input", "02022020");
})();
```

I get an exception `Error: Cannot fill input of type "date".`

**Comment by aslushnikov** [comment]
Looks like this is fixed with #1676
